posts/2022-08-11-Rust-1.63.0.md

```diff
@@ -0,0 +1,143 @@
+`#[repr(transparent)]`, meaning that `extern "C"` bindings can directly take
+these types to encode the ownership semantics. See the stabilized APIs section
+for the full list of wrapper types stabilized in 1.63, currently, they are
+available on cfg(unix) platforms, Windows, and WASI.
+
+We recommend that new APIs use these types instead of the previous type aliases
+(like [`RawFd`]).
+
+[`RawFd`]: https://doc.rust-lang.org/stable/std/os/unix/io/type.RawFd.html
+[`BorrowedFd`]: https://doc.rust-lang.org/stable/std/os/unix/io/struct.BorrowedFd.html
+[`OwnedFd`]: https://doc.rust-lang.org/stable/std/os/unix/io/struct.OwnedFd.html
+
+### Turbofish for generics in functions with `impl Trait`
+
+The turbofish notation (`foo::<u32>()`) can be used to specify the types for explicit type
+parameters (`fn foo<T: Copy>`) even when `impl Trait` is used in the argument types.
+Specifying the type of an `impl Trait` argument via turbofish is still not possible.
+
+### Non-lexical lifetimes migration complete
+
+As detailed in [this blog post], we've fully removed the previous lexical borrow checker
+from rustc across all editions, fully enabling the non-lexical, new, version of the borrow
+checker. Since the borrow checker doesn't affect the output of rustc, this won't change
+the behavior of any programs, but it completes a long-running migration (started in the
+initial stabilization of NLL for the 2018 edition) to deliver the full benefits of the new
+borrow checker across all editions of Rust. For most users, this change will bring
+slightly better diagnostics for some borrow checking errors, but will not otherwise impact
+which code they can write.
+
+You can read more about non-lexical lifetimes in [this section of the 2018 edition announcement][nll].
+
+[this blog post]: https://github.com/rust-lang/blog.rust-lang.org/pull/989
+[nll]: https://blog.rust-lang.org/2018/12/06/Rust-1.31-and-rust-2018.html#non-lexical-lifetimes
+
+### Stabilized APIs
+
+The following methods and trait implementations are now stabilized:
+
+- [`array::from_fn`]
+- [`Box::into_pin`]
+- [`BinaryHeap::try_reserve_exact`]
+- [`BinaryHeap::try_reserve`]
+- [`OsString::try_reserve`]
+- [`OsString::try_reserve_exact`]
+- [`PathBuf::try_reserve`]
+- [`PathBuf::try_reserve_exact`]
+- [`Path::try_exists`]
+- [`Ref::filter_map`]
+- [`RefMut::filter_map`]
+- [`NonNull::<[T]>::len`]
+- [`ToOwned::clone_into`]
+- [`Ipv6Addr::to_ipv4_mapped`]
+- [`unix::io::AsFd`]
+- [`unix::io::BorrowedFd<'fd>`]
+- [`unix::io::OwnedFd`]
+- [`windows::io::AsHandle`]
+- [`windows::io::BorrowedHandle<'handle>`]
+- [`windows::io::OwnedHandle`]
+- [`windows::io::HandleOrInvalid`]
+- [`windows::io::HandleOrNull`]
+- [`windows::io::InvalidHandleError`]
+- [`windows::io::NullHandleError`]
+- [`windows::io::AsSocket`]
+- [`windows::io::BorrowedSocket<'handle>`]
+- [`windows::io::OwnedSocket`]
+- [`thread::scope`]
+- [`thread::Scope`]
+- [`thread::ScopedJoinHandle`]
```

Comment:
I would like to suggest that the scoped threads stabilisation is called out in the release notes. Scoped threads make it possible to share (borrow) data across threads without runtime synchronisation which isn't possible with regular threads, and it's a big deal that they're available out of the box in the std library in a stable release!

Reply:
Done! Would appreciate feedback on the exact language; no strong feelings about it.